### ISSUE TITLE
QEMU/SeaBIOS identifiers detectable

### ISSUE THREAD

**Issue by ChrisThibodeaux** [kevoreilly/CAPEv2#2413]
## About accounts on [capesandbox.com](https://capesandbox.com/)
* Issues isn't the way to ask for account activation. Ping capesandbox in [Twitter](https://twitter.com/capesandbox) with your username

## This is open source and you are getting __free__ support so be friendly!

# Prerequisites

Please answer the following questions for yourself before submitting an issue.

- [x] I am running the latest version
- [x] I did read the README!
- [x] I checked the documentation and found no answer
- [x] I checked to make sure that this issue has not already been filed
- [x] I'm reporting the issue to the correct repository (for multi-repository projects)
- [x] I have read and checked all configs (with all optional parts)

# Current Behavior

**Not considering this a bug**, could be an issue with my setup. Followed the docs, editing and running the install scripts. Followed the two KVM posts from @doomedraven to get everything set as well.

Identifiers for QEMU and SeaBIOS present on guest machines. Here are examples from the registry:
```
HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Enum\ACPI\QEMU0002\3&11583659&0
HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Enum\ACPI\QEMU0002
HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Enum\ACPI\QEMU0002\3&11583659&0\Device Parameters
HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Enum\ACPI\QEMU0002\3&11583659&0
HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Enum\ACPI\QEMU0002\3&11583659&0\LogConf
HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Enum\ACPI\QEMU0002\3&11583659&0

HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\Class\{4d36e968-e325-11ce-bfc1-08002be10318}\0000\HardwareInformation.ChipType = SeaBIOS VBE(C) 2011
HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\Class\{4d36e968-e325-11ce-bfc1-08002be10318}\0000\HardwareInformation.DacType = SeaBIOS Developers
HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\Class\{4d36e968-e325-11ce-bfc1-08002be10318}\0000\HardwareInformation.BiosString = SeaBIOS VBE Adapter
HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\Class\{4d36e97d-e325-11ce-bfc1-08002be10318}\0035\HardwareInformation.ChipType = SeaBIOS VBE(C) 2011
HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\Class\{4d36e97d-e325-11ce-bfc1-08002be10318}\0035\HardwareInformation.DacType = SeaBIOS Developers
HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\Class\{4d36e97d-e325-11ce-bfc1-08002be10318}\0035\HardwareInformation.BiosString = SeaBIOS VBE Adapter
```

1. `ACPI\QEMU0002\3&11583659&0` is a known artifact [found on Virus Total and Hybrid Analysis VMs](https://obbedcode.hashnode.dev/bad-honey-in-the-honey-pot) (search for `QEMU0002` in that article). This shows under device manager as `Other devices\Unknown Device` and has set registry keys with the identifying string.
2. The various `SeaBIOS` are also found in the above article and used by [AsyncRAT in the wild](https://www.huntress.com/blog/fake-browser-updates-lead-to-boinc-volunteer-computing-software).

![seabiod_video_indicators](https://github.com/user-attachments/assets/c3829938-17d0-4380-aba8-c2beff8fe49d)
![regi_keys](https://github.com/user-attachments/assets/070ff748-ae47-4b5b-876c-801e3f4abf20)

I know that the AsyncRAT check is at least accounted for by community YARA rule for VenomRAT. Not sure if anything is in place for the `ACPI\QEMU0002\3&11583659&0` one, but that might just be an issue with my install? I've uninstalled the device and removed the registry keys. For the time being, I am manually editing the keys that have any SeaBIOS in their data.

Looking for advice on this. Should I update the `kvm-qemu.sh` to cover the SeaBIOS strings in [vgasrc/vbe.c](https://github.com/coreboot/seabios/blob/62a1429ec1ec67f14c039d97627a6a7ef70a983c/vgasrc/vbe.c#L19)?

**Comment by doomedraven** [comment]
Hey yes, you need to extend it, I have my private version that I use, so I don't use public one, public is an examples that I did for community with base fixes, but there is way much more to fix. But as you understand I can't show all of them otherwise those who implements detection will have advantage. So you need to fix that by youself

**Comment by ChrisThibodeaux** [comment]
Okay, that makes perfect sense. I did not consider that, should I edit my original post to redact those items?

**Comment by doomedraven** [comment]
Bah thats fine :) better than edit, you can PR fix to kvm-qemu ;) 

In most of the cases yes, but from my experience, this specific qemu002 need recreation of vm, but true to fix, reboot vm, if it doesn't disappear then you will need to recreate OR use psexec to force dump of registry, edit and restore patched version,  see disguise.py there should be example how we were doing that on live machine

**Comment by doomedraven** [comment]
feel free to post/ask here, but closing as resolved

**Comment by ChrisThibodeaux** [comment]
@doomedraven Thanks! CAPE was and has been my only source of experience with sandboxing and dynamic analysis. I am learning more about malware through CAPE and more about CAPE through malware. I love the sandbox and the project as a whole. Incredible stuff all around and I'll contribute as much as I can, wherever I can. I'm working on the updates to the `kvm-qemu.sh` now and setting up a laptop as a dedicated build test environment.

I was wondering about the idea of continuously hardening VMs versus creating hooks to handle anti-vm techniques. They both have their place, but after a certain point of hardening, I don't know if it makes sense to continue. It seems like the flexibility of being able to enable/disable hooks gives you more variability to the run environment. Hardening is sort of set in stone, unable to vary between one run and the next for a given VM.

Where you draw the line, if at all, and stop hardening a VM against detection?

**Comment by doomedraven** [comment]
I also learned a lot since 2011 when I started with cuckoo/cuckoo-modified/cape. Glad you learning a lot. In my personal point of view, is better to handle on source code level so you don't need to care about stuff inside of the VM, as each hook is huge delay for the performance. And when more hooks we add, bad guys can start abusing that to defeat analysis. Qemu introduces new patterns rarely so is easier to check if there is new issue to fix.

about the line, well they do another things to defeat analysis, i see less and less malware care about vm stuff, as is easier to spot in server side if is real pc or sandbox. is complicated subject and mostly depends of kind of malware that you focus on
